### ISSUE TITLE
False positive with variable in `no-unused-message-ids`

### ISSUE THREAD

**Issue by bmish** [not-an-aardvark/eslint-plugin-eslint-plugin#275]
In this example, we're failing to see that `messageId` from the variable is used.

This issue is not specific to TypeScript.

```ts
function checkCall(
  node: TSESTree.Node,
  reportingNode: TSESTree.Node,
  messageId: MessageIds,
): void {
  if (util.isTypeAnyType(type)) {
    if (/* ... */) {
      messageId = 'unsafeCallThis';
    }
    context.report({
      node: reportingNode,
      messageId: messageId,
    });
  }
}
```

Example: https://github.com/typescript-eslint/typescript-eslint/blob/ad412cdd74dc5619fbe4bf27c0a5eb5c5a4b65ca/packages/eslint-plugin/src/rules/no-unsafe-call.ts#L68

**Comment by bmish** [comment]
Turns out I misinterpreted what was happening. There's no problem here.
